py/webdriver/edge/webdriver.py

```diff
@@ -0,0 +1,203 @@
+           capabilities only, such as "proxy" or "loggingPref".
+         - port - port you would like the service to run, if left as 0, a free port will be found.
+         - verbose - whether to set verbose logging in the service
+         - service_log_path - Where to log information from the driver.
+         - log_path: Deprecated argument for service_log_path
+         - keep_alive - Whether to configure EdgeRemoteConnection to use HTTP keep-alive.
+         - desired_capabilities - Dictionary object with non-browser specific
+           capabilities only, such as "proxy" or "loggingPref".
+         - service_args - List of args to pass to the driver service
+         - options - this takes an instance of EdgeOptions
+         - edge_options - Deprecated argument for options
+         - use_chromium - Whether to use chromium edge driver, default to use legacy edge driver
+         """
+        self.use_chromium = use_chromium
+
+        if edge_options:
+            warnings.warn('use options instead of edge_options',
+                          DeprecationWarning, stacklevel=2)
+            options = edge_options
+
+        # options use_chromium flag validation
+        if self.use_chromium and options and not options.use_chromium:
+            raise Exception(
+                'options.use_chromium must be set to true when using an Edge Chromium driver service.')
+        if not self.use_chromium and options and options.use_chromium:
+            raise Exception(
+                'options.use_chromium must be set to false when using an Edge Legacy driver service.')
+
+        if self.use_chromium:
+            if options is None:
+                # desired_capabilities stays as passed in
+                if desired_capabilities is None:
+                    desired_capabilities = self.create_options().to_capabilities()
+            else:
+                if desired_capabilities is None:
+                    desired_capabilities = options.to_capabilities()
+                else:
+                    desired_capabilities.update(options.to_capabilities())
+
+        if log_path:
+            warnings.warn('use service_log_path instead of log_path',
+                          DeprecationWarning, stacklevel=2)
+            service_log_path = log_path
+
+        self.port = port
+        if self.port == 0:
+            self.port = utils.free_port()
+
+        if self.use_chromium:
+            self.service = Service(
+                executable_path,
+                port=port,
+                verbose=verbose,
+                service_args=service_args,
+                log_path=service_log_path)
+            self.service.start()
+        else:
+            self.edge_service = Service(
+                executable_path, 
+                port=self.port, 
+                verbose=verbose, 
+                log_path=service_log_path)
+            self.edge_service.start()
+
+        try:
+            if self.use_chromium:
```

Comment:
Again, for simplicity, let's just handle the RemoteWebDriver call one way for both legacy and chromium mode. The resolve_ip param isn't actually needed for legacy Edge anymore. It was fixed in a previous Windows 10 update. So it should be safe to just use the chromium version of this call.